### PR TITLE
Add linting gha

### LINKED ISSUE #192: GHA Linting Check
From [this comment](https://github.com/SenteraLLC/ulabel/pull/175#pullrequestreview-2346761919).
Upon merging #175, we'll have some minimal formatting and linting rule enforcement.

A Github Action workflow to check linting/formatting prior to merge would be nifty.

### PR DESCRIPTION
**Pull request by TrevorBurgoyne**
# Linting Github Action

## Description

- Closes #192 

## PR Checklist

- [x] Merged latest main
- [x] Version number in `package.json` has been bumped since last release
- [x] Version numbers match between package `package.json` and `src/version.js`
- [x] Ran `npm install` and `npm run build` AFTER bumping the version number
- [x] Updated documentation if necessary (currently just in `api_spec.md`)
- [x] Added changes to `changelog.md`

## Breaking API Changes
no

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Lint
+
+on: [push, pull_request]
+
```
Comment (joshua-dean) [inline .github/workflows/lint.yml:4]: Would it be better to do pushes to `main`, PRs, and then the `workflow_dispatch` for manual kick-offs?
Might burn a bunch of unnecessary compute running on every push, at the cost of requiring someone to manually run them periodically while reviewing a PR.

Comment (joshua-dean) [reply .github/workflows/lint.yml:4]: Regardless I think `workflow_dispatch` might be useful if we ever need to test changes by running from new feature branches.

Comment (TrevorBurgoyne) [reply .github/workflows/lint.yml]: resolved

**Review by joshua-dean — APPROVED** [review]
Cool, we can always change it later if manual kickoff becomes a hassle

**Pull request merged**